packages/kbn-test-jest-helpers/src/axe_helpers.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { axe } from 'jest-axe';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { ImpactValue, Result } from 'axe-core';
+import { ReactWrapper } from './testbed/types';
+
+/**
+ * Function to get an array of a11y violations from axe automated testing
+ * @param component
+ * @param impactLevel One of 'minor', 'moderate', 'serious', 'critical'. Default is 'critical'.
+ */
+export const getA11yViolations = async (
+  component: ReactWrapper,
+  impactLevel?: ImpactValue
+): Promise<Result[]> => {
+  impactLevel = impactLevel ?? 'critical';
+  const axeResults = await axe(component.html(), { resultTypes: ['violations'] });
+  return axeResults.violations.filter((violation) => violation.impact === impactLevel);
+};
+
+/**
+ * Function to test if a component doesn't have a11y violations from axe automated testing
+ * @param component
+ * @param impactLevel One of 'minor', 'moderate', 'serious', 'critical'. Default is 'critical'.
+ */
+export const expectToBeAccessible = async (
+  component: ReactWrapper,
+  impactLevel?: ImpactValue
+): Promise<void> => {
+  const violations = await getA11yViolations(component, impactLevel);
+  expect(violations).toHaveLength(0);
```

Comment:
Will the output for this include all of the violations?

Reply:
That is a great question! I think I need to update the logic here, because it will only get 'critical' violations by default. I don't even think I need a parameter for `impactLevel` here. I'll ask in the a11y channel which level we want to test for.

Reply:
Great call removing the `impactLevel`. Our goal is to find and remove all accessibility issues, regardless of their impact. It has been my experience that serious and moderate issues can cause significant UX problems in large enough quantities. They're not breaking UX issues per se, but each one causes friction and adds to the accumulated cognitive load.